Core.Collectors/Web/HttpUtility.cs

```diff
@@ -19,6 +22,22 @@ public static async Task<JObject> ParseAsJObjectAsync(HttpResponseMessage respon
             return JObject.Load(jsonTextReader);
         }
 
+        public static async Task<Tuple<bool, JObject>> TryParseAsJObjectAsync(HttpResponseMessage response)
+        {
+            using Stream responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            // Loading a stream / text into JObject expands ~10x in memory. In other words, a 100MB response, will approx. take 1GB ram.
+            // Our machines (where this is a problem --- for CG collection) have 16 GB ram. I am leaving 4 GB for OS (might not be needed but just in case) and the rest of the code.
+            // That leaves 12 GB to be shared between 6, or 2 GB each, so ~200 MB raw response.
+            if (responseStream.Length > 200 * MB)
```

Comment:
What about making the max size a default parameter to the function?

Reply:
What do you have in mind? Do you want to also make this customizable at the collection node level?

Reply:
Not much actually. It was just a thought, why have this limit hardcoded in the function? Making this a parameter makes it easier to change if we need to -- let's say, we need a 250 MiB limit or 200 is already too much. No need to mess with the submodule then.

Reply:
Fair point, did not think about hardcoding in submodule vs. making it easily change-able on the caller end. WIll change.